payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CardNumberController.kt

```diff
@@ -57,8 +57,13 @@ internal class CardNumberController constructor(
         CardBrand.getCardBrands(it).firstOrNull() ?: CardBrand.Unknown
     }
 
-    override val trailingIcon: Flow<TextFieldIcon?> = cardBrandFlow.map {
-        TextFieldIcon(it.icon, isIcon = false)
+    override val trailingIcon: Flow<TextFieldIcon?> = _fieldValue.map {
+        val cardBrands = CardBrand.getCardBrands(it)
```

Comment:
Does this account for the card metadata service?

Reply:
Does it need to?

Reply:
Couldn't the card metadata service return a value different from our hard coded brands?

Reply:
Can you share how you resolved this?

Reply:
If the accountRangeService returned a cardbrand we use that one